### ISSUE TITLE
Unicode character in image filename makes sphinx crash

### ISSUE THREAD

**Issue by Konfuz** [sphinx-doc/sphinx#2395]
## How to reproduce

Take an image, name it, for example, `testimäge.jpg` and include it to your documentation using the `.. image::` directive.

The following will happen when you build the documentation:

```
make html
sphinx-build -b html -d _build/doctrees   . _build/html
Running Sphinx v1.4
loading translations [de]... done
loading pickled environment... done
building [mo]: targets for 0 po files that are out of date
building [html]: targets for 1 source files that are out of date
updating environment: 0 added, 1 changed, 0 removed
reading sources... [100%] index                                                                    
Encoding error:
'ascii' codec can't encode character u'\xe4' in position 33: ordinal not in range(128)
The full traceback has been saved in /tmp/sphinx-err-8cH_lS.log, if you want to report the issue to the developers.
make: *** [html] Error 1
```

This used to work in Sphinx 1.3 but is broken in 1.4 
## Error location

The Error is related to **line 203** in **sphinx.util.i18n**

``` python
def get_image_filename_for_language(filename, env):
    if not env.config.language:
        return filename

    filename_format = env.config.figure_language_filename
    root, ext = path.splitext(filename)
    try:
        return filename_format.format(root=root, ext=ext,
                                      language=env.config.language)
    except KeyError as exc:
        raise SphinxError('Invalid figure_language_filename: %r' % exc)
```

where both an ASCII and unicode string are used in an format, which in turn makes sphinx exit involuntarily. 
## Possible solution

One could either edit the **sphinx.config line 61** to say

``` python
figure_language_filename = (u'{root}.{language}{ext}', 'env', [str]),
```

instead of

``` python
figure_language_filename = ('{root}.{language}{ext}', 'env', [str]),
```

or adapt the aforementioned utility-function to make sure all strings are unicode.
I don't know which one would be better, hence no pull-request.

**Comment by shimizukawa** [comment]
Sorry for your inconvenience. Thanks for the inspection, I got a point. I'll fix it ASAP.

**Comment by tk0miya** [comment]
Fixed at 9df7783
Thank you for reporting

**Comment by Konfuz** [comment]
Thank you for fixing it (and in reasonably short time as well)!
